### PR TITLE
Fix Parser.parseDocuments

### PR DESCRIPTION
**Pull request by sideeffffect**
fixes https://github.com/circe/circe-yaml/issues/343
closes https://github.com/circe/circe-yaml/pull/342
/cc @lucaviolanti

### CODE REVIEW

File: circe-yaml-v12/src/test/scala/io/circe/yaml/v12/ParserTests.scala [hunk 1]
```diff
@@ -121,4 +123,132 @@ class ParserTests extends AnyFlatSpec with Matchers with EitherValues {
...
+    assert(result.head.isLeft)
+  }
+
+  it should "fail to parse invalid YAML" in {
+    /*
+    See comments below that document the actual behaviour
+     */
+    // val result = Either.catchNonFatal(parser.parseDocuments(new StringReader("""foo: - bar""")).toList)
+    // assert(result.isLeft)
+    // assert(result.swap.value.isInstanceOf[org.yaml.snakeyaml.scanner.ScannerException])
```
Comment (lucaviolanti) [inline circe-yaml-v12/src/test/scala/io/circe/yaml/v12/ParserTests.scala:150]:
```suggestion
```

File: circe-yaml-v12/src/test/scala/io/circe/yaml/v12/ParserTests.scala [hunk 2]
```diff
@@ -121,4 +123,132 @@ class ParserTests extends AnyFlatSpec with Matchers with EitherValues {
...
+  it should "parse decimal with underscore breaks as strings" in {
+    val result = parser.parseDocuments(new StringReader("""foo: 1_000_000""")).toList
+    assert(result.size == 1)
+    assert(result.head.contains(Map("foo" -> "1_000_000").asJson))
+  }
+
+  it should "parse empty string as 0 documents" in {
+    /*
+    See comments below that document the actual behaviour
+     */
```
Comment (lucaviolanti) [inline circe-yaml-v12/src/test/scala/io/circe/yaml/v12/ParserTests.scala:178]:
```suggestion
```

File: circe-yaml-v12/src/test/scala/io/circe/yaml/v12/ParserTests.scala [hunk 3]
```diff
@@ -121,4 +123,132 @@ class ParserTests extends AnyFlatSpec with Matchers with EitherValues {
...
+    val result = parser.parseDocuments(new StringReader("")).toList
+    assert(result.isEmpty)
+  }
+
+  it should "parse blank string as 0 documents" in {
+    /*
+    See comments below that document the actual behaviour
+     */
```
Comment (lucaviolanti) [inline circe-yaml-v12/src/test/scala/io/circe/yaml/v12/ParserTests.scala:186]:
```suggestion
```

File: circe-yaml-v12/src/test/scala/io/circe/yaml/v12/ParserTests.scala [hunk 4]
```diff
@@ -121,4 +123,132 @@ class ParserTests extends AnyFlatSpec with Matchers with EitherValues {
...
+    //             | baz:
+    //             |  - *alias1
+    //             |  - *alias1
+    //             |""".stripMargin
+    //       )
+    //     )
+    //     .toList
+    // )
+    // assert(result.isLeft)
+    // assert(result.swap.value.isInstanceOf[org.yaml.snakeyaml.error.YAMLException])
```
Comment (lucaviolanti) [inline circe-yaml-v12/src/test/scala/io/circe/yaml/v12/ParserTests.scala:233]:
```suggestion
```

File: circe-yaml/src/test/scala/io/circe/yaml/ParserTests.scala [hunk 5]
```diff
@@ -122,4 +123,131 @@ class ParserTests extends AnyFlatSpec with Matchers with EitherValues {
...
+    assert(result.head.isLeft)
+  }
+
+  it should "fail to parse invalid YAML" in {
+    /*
+    See comments below that document the actual behaviour
+     */
+    // val result = Either.catchNonFatal(parser.parseDocuments(new StringReader("""foo: - bar""")).toList)
+    // assert(result.isLeft)
+    // assert(result.swap.value.isInstanceOf[org.yaml.snakeyaml.scanner.ScannerException])
```
Comment (lucaviolanti) [inline circe-yaml/src/test/scala/io/circe/yaml/ParserTests.scala:150]:
```suggestion
```

File: circe-yaml/src/test/scala/io/circe/yaml/ParserTests.scala [hunk 6]
```diff
@@ -122,4 +123,131 @@ class ParserTests extends AnyFlatSpec with Matchers with EitherValues {
...
+    See comments below that document the actual behaviour
+     */
+    val result = parser.parseDocuments(new StringReader("")).toList
+    assert(result.isEmpty)
+  }
+
+  it should "parseDocuments blank string as 0 documents" in {
+    /*
+    See comments below that document the actual behaviour
+     */
```
Comment (lucaviolanti) [inline circe-yaml/src/test/scala/io/circe/yaml/ParserTests.scala:186]:
```suggestion
```

File: circe-yaml/src/test/scala/io/circe/yaml/ParserTests.scala [hunk 7]
```diff
@@ -122,4 +123,131 @@ class ParserTests extends AnyFlatSpec with Matchers with EitherValues {
...
+    //             | baz:
+    //             |  - *alias1
+    //             |  - *alias1
+    //             |""".stripMargin
+    //       )
+    //     )
+    //     .toList
+    // )
+    // assert(result.isLeft)
+    // assert(result.swap.value.isInstanceOf[org.yaml.snakeyaml.error.YAMLException])
```
Comment (lucaviolanti) [inline circe-yaml/src/test/scala/io/circe/yaml/ParserTests.scala:233]:
```suggestion
```

File: circe-yaml/src/test/scala/io/circe/yaml/ParserTests.scala [hunk 8]
```diff
@@ -122,4 +123,131 @@ class ParserTests extends AnyFlatSpec with Matchers with EitherValues {
...
+  it should "parse decimal with underscore breaks" in {
+    val result = parser.parseDocuments(new StringReader("""foo: 1_000_000""")).toList
+    assert(result.size == 1)
+    assert(result.head.contains(Map("foo" -> 1000000).asJson))
+  }
+
+  it should "parseDocuments empty string as 0 documents" in {
+    /*
+    See comments below that document the actual behaviour
+     */
```
Comment (lucaviolanti) [inline circe-yaml/src/test/scala/io/circe/yaml/ParserTests.scala:178]:
```suggestion
```

**Review by lucaviolanti — CHANGES_REQUESTED** [review]

**Review by lucaviolanti — APPROVED** [review]

**Review by zarthross — APPROVED** [review]
I think this looks fine.  I'll merge it in after https://github.com/circe/circe-yaml/pull/348 so we can start running more thorough CI on PRs.

**Pull request merged**